select.go

```diff
@@ -288,6 +297,33 @@ func (b SelectBuilder) RightJoin(join string, rest ...interface{}) SelectBuilder
 	return b.JoinClause("RIGHT JOIN "+join, rest...)
 }
 
+// PreWhere adds an expression to the PREWHERE clause of the query.
+//
+// Expressions are ANDed together in the generated SQL.
+//
+// PreWhere accepts several types for its pred argument:
+//
+// nil OR "" - ignored.
+//
+// string - SQL expression.
+// If the expression has SQL placeholders then a set of arguments must be passed
+// as well, one for each placeholder.
+//
+// map[string]interface{} OR Eq - map of SQL expressions to values. Each key is
+// transformed into an expression like "<key> = ?", with the corresponding value
+// bound to the placeholder. If the value is nil, the expression will be "<key>
+// IS NULL". If the value is an array or slice, the expression will be "<key> IN
+// (?,?,...)", with one placeholder for each item in the value. These expressions
+// are ANDed together.
+//
+// PreWhere will panic if pred isn't any of the above types.
+func (b SelectBuilder) PreWhere(pred interface{}, args ...interface{}) SelectBuilder {
+	if pred == nil || pred == "" {
+		return b
+	}
+	return builder.Append(b, "PreWhereParts", newWherePart(pred, args...)).(SelectBuilder)
```

Comment:
nit: comment that it's safe to reuse newWherePart() here as prewhere and where clauses expect same type of predicates.

Reply:
This is already done as part of the existing `Having` implementation so probably not necessary to point it out.